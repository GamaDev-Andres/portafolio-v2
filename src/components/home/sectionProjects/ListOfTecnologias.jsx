import PropTypes from 'prop-types'

const ListOfTecnologias = ({ tecnologias }) => {
  return (
    <div>
      <h3 className="text-letra font-medium dark:text-letraDark">
        Tecnologias implementadas
      </h3>
      <ul className="flex gap-2 flex-wrap py-2">
        {tecnologias.map((el) => (
          <li key={el} className="px-2 py-1 border border-bordes rounded-full">
            <span>{el}</span>
          </li>
        ))}
      </ul>
    </div>
  )
}
ListOfTecnologias.propTypes = {
  tecnologias: PropTypes.array.isRequired
}
export default ListOfTecnologias
