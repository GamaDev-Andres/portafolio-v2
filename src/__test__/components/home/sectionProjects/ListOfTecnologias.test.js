/**
 * @jest-environment jsdom
 */
import Enzyme from 'enzyme'
import ListOfTecnologias from '../../../../components/home/sectionProjects/ListOfTecnologias'

describe('probando <ListOfTecnologias/>', () => {
  const tecnologias = ['react', 'js', 'css', 'html']
  const wrapper = Enzyme.mount(
    <ListOfTecnologias tecnologias={tecnologias} />
  )
  test('debe montarse correctamente', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('h3').text().trim()).toBe('Tecnologias implementadas')
    expect(wrapper.find('li').length).toBe(tecnologias.length)
  })
})
