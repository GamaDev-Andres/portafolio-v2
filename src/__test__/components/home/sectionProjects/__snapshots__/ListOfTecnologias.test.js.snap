// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probando <ListOfTecnologias/> debe montarse correctamente 1`] = `
<div>
  <h3
    className="text-letra font-medium dark:text-letraDark"
  >
    Tecnologias implementadas
  </h3>
  <ul
    className="flex gap-2 flex-wrap py-2"
  >
    <li
      className="px-2 py-1 border border-bordes rounded-full"
      key="react"
    >
      <span>
        react
      </span>
    </li>
    <li
      className="px-2 py-1 border border-bordes rounded-full"
      key="js"
    >
      <span>
        js
      </span>
    </li>
    <li
      className="px-2 py-1 border border-bordes rounded-full"
      key="css"
    >
      <span>
        css
      </span>
    </li>
    <li
      className="px-2 py-1 border border-bordes rounded-full"
      key="html"
    >
      <span>
        html
      </span>
    </li>
  </ul>
</div>
`;
